one_step_setup.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+set -e
+
+function fail {
+    # Print a newline to stderr and exit.
+    echo > /dev/stderr
+    exit 1
+}
+
+# Make sure we don't override an existing Pants installation.
+if [ -f "pants.toml" ]
+then
+    echo A \`pants.toml\` file already exists in this directory. This installation script \
+        needs to create a new \`pants.toml\` file. > /dev/stderr
+    echo If you are trying to upgrade to a newer version of Pants, set the \`pants_version\` \
+        setting in your existing \`pants.toml\` file. > /dev/stderr
+    echo If you are trying to run the install process from scratch, delete the \`pants.toml\` \
+        file and start again. > /dev/stderr
+    fail 
+fi
+
+if [ -f "pants" ]
+then
+    echo A file called \`pants\` already exists in this directory. This installation script \
+        will overwrite this file. To proceed, either rename this file, or delete the \
+        \`pants\` file. > /dev/stderr
+    fail
+fi
+
+# Find the latest stable version from PyPI if not set at the command line.
+if [ -z "$PANTS_VERSION" ]
+then
+    PANTS_VERSION=`curl https://pypi.org/pypi/pantsbuild.pants/json | \
+        grep -o '"version":"[^"]*"' | \
+        grep -o "[0-9]*\\.[0-9]*\\.[0-9]*"`
+fi
+
+# Create enough of a pants.toml file that our bootstrap process can run
+printf '[GLOBAL]\npants_version = "'$PANTS_VERSION'"\n' > pants.toml
+
+# Fetch the Pants bootstrap script and run it to verify that we fetched
+# the correct version
+curl -L -O https://static.pantsbuild.org/setup/pants 
```

Comment:
It's not unreasonable. It _would_, however, mean we'd need to keep the one-step-setup script up to date whenever we  update the `pants` script. We'd catch that in the integration tests, but it'd be tedious.

Reply:
A bit tedious, but a small extra maintenance burden seems worth the better security. 

I cced @benjyw because we both have a lot of concern about supply chain attacks after reading https://www.amazon.com/This-They-Tell-World-Ends/dp/1635576059

Reply:
I'm not worried about the tedium, I am worried about the setup script failing for users because we forgot to update the sha, even if we eventually catch this in a test.

Reply:
How could that ever happen? The test would fail and block us from merging changes.

Reply:
So as written, and as how our tests currently work, it will be _quite_ tedious to check that `one_step_setup.sh` actually works in a test. That's because the script fetches from `https://static.pantsbuild.org`, rather than the newly updated version of the script.

That means, we either need to spawn a web server in the tests (not insurmountable, just tedious) and hope that the integration test still succeeds when we stop using a false server (so the integration test is less trustworthy); or we need to do a two-step PR process when we update `./pants` (once to update `pants` and once to update `one_step_setup.sh`).

We may want to follow up on this one later, if you ask me.

Reply:
Ah okay, good point.